packages/babel-traverse/src/scope/index.ts

```diff
@@ -833,7 +833,13 @@ class Scope {
 
         // A redeclaration of an existing variable is a modification
         if (local) {
-          this.registerConstantViolation(bindingPath);
+          // this.registerConstantViolation(bindingPath);
+          const ids = path.getBindingIdentifiers();
```

Comment:
Maybe we can use the `ids` above here, and the replacement will pass all tests in our repository.
The difference is `getOuterBindingIdentifiers` vs `getBindingIdentifiers`. It seems that the first one is more appropriate to use here?

Reply:
Yes. We should reuse the `ids` above, and we don't have to loop through the binding identifiers again, we can just call `local.reassign(bindingPath)`.